### PR TITLE
add river/layout

### PR DESCRIPTION
**Pull request by alex-courtis**
Similar functionality to river/window but shows layout.

Updated river output status protocol 2->4

Tested OK:
* hotplug monitor
* river output status version 2 - module does not load
* css classes
* all configuration options

### CODE REVIEW

File: src/modules/river/layout.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    spdlog::error("river_status_manager_v1 not advertised");
+    return;
+  }
+
+  if (!seat_) {
+    spdlog::error("wl_seat not advertised");
+  }
+
+  label_.hide();
+  ALabel::update();
```
Comment (alex-courtis) [inline src/modules/river/layout.cpp:124]: This does not initially hide the label. I did notice the same behaviour from window... any ideas?

**Comment by alex-courtis** [comment]
It looks like there was an infrastructural issue with the builds. Can I re-run the builds?

Should I push a dummy commit to force a re-run?

**Comment by alex-courtis** [comment]
Merged master and updated to obey `cpp_std=c++20`

**Comment by alex-courtis** [comment]
I'll add to the wiki after merge.

File: src/modules/river/layout.cpp (hunk 1 above)
Comment (Alexays) [reply src/modules/river/layout.cpp:124]: Maybe because the label isn't empty, and so ALabel::update re-show it 🤔

**Comment by pinpox** [comment]
@alex-courtis How do I make this work with `river-luatile`? Where do I have to specify the layout name?

**Comment by alex-courtis** [comment]
> @alex-courtis How do I make this work with `river-luatile`? Where do I have to specify the layout name?

It's up to `river-luatile` to specify the layout name each time it pushes/commits.

The name seems to be specified here:
https://github.com/MaxVerevkin/river-luatile/blob/0ae8e1ca68417878674469def3c861a148fac867/src/main.rs#L55

**Pull request merged**